app/markets/ActivityTable.tsx

```diff
@@ -31,64 +33,107 @@ const getTagColorScheme = (transactionType: TransactionType): TagColorSchemeProp
   }
 };
 
+const ITEMS_PER_PAGE = 10;
+
 export const ActivityTable = ({ id }: { id: string }) => {
+  const [page, setPage] = useState(1);
+
   const { data, isLoading } = useQuery({
-    queryKey: ['getMarketTransactions', id],
+    queryKey: ['getMarketTransactions', id, page - 1],
+    queryFn: async () =>
+      getMarketTransactions({
+        first: ITEMS_PER_PAGE,
+        skip: (page - 1) * ITEMS_PER_PAGE,
+        id: id.toLowerCase(),
+      }),
+  });
+
+  const { data: transactionsNextPage } = useQuery({
+    queryKey: ['getMarketTransactions', id, page],
     queryFn: async () =>
       getMarketTransactions({
-        first: 20,
+        first: ITEMS_PER_PAGE,
+        skip: page * ITEMS_PER_PAGE,
         id: id.toLowerCase(),
       }),
   });
 
+  const hasMoreMarkets =
+    transactionsNextPage && transactionsNextPage.fpmmTransactions.length !== 0;
```

Comment:
Minor, non-blocking

```suggestion
    transactionsNextPage && transactionsNextPage.fpmmTransactions.length;
```

Reply:
I find `!==0` more readable. That way I can easily understand that it isn't an empty array.